contracts/src/common/MigrationController.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.13;
+
+import {IBaseRegistrar} from "@ens/contracts/ethregistrar/IBaseRegistrar.sol";
+import {INameWrapper, CANNOT_UNWRAP} from "@ens/contracts/wrapper/INameWrapper.sol";
+import {IERC1155Receiver} from "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import {ERC165, IERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {IMigrationStrategy} from "./IMigration.sol";
+import {TransferData, MigrationData} from "./TransferData.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title MigrationController
+ * @dev Base contract for the v1-to-v2 migration controller.
+ */
+abstract contract MigrationController is IERC1155Receiver, IERC721Receiver, ERC165, Ownable {
+    error CallerNotEthRegistryV1(address caller);   
+    error NoMigrationStrategySet();
+    error MigrationFailed();
+    error NotOwner(uint256 tokenId);
+
+    event StrategySet(IMigrationStrategy strategy);
+
+    IBaseRegistrar public immutable ethRegistryV1;
+    IMigrationStrategy public strategy;
+
+    constructor(IBaseRegistrar _ethRegistryV1) Ownable(msg.sender) {
+        ethRegistryV1 = _ethRegistryV1;
+    }
+
+    /**
+     * @dev Sets the migration strategy.
+     *
+     * @param _strategy The migration strategy.
+     */
+    function setStrategy(IMigrationStrategy _strategy) external onlyOwner {
+        strategy = _strategy;
+        emit StrategySet(_strategy);
+    }
+
+    /**
+     * Implements ERC165.supportsInterface
+     */
+    function supportsInterface(bytes4 interfaceId) public virtual view override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(MigrationController).interfaceId || interfaceId == type(Ownable).interfaceId || interfaceId == type(IERC1155Receiver).interfaceId || interfaceId == type(IERC721Receiver).interfaceId || super.supportsInterface(interfaceId);
```

Comment:
Can you format this with each condition on a line? Feel free to use a linter/prettifier instead.